sequencers/modeshape-sequencer-video/src/main/java/org/modeshape/sequencer/video/VideoMetadataSequencer.java

```diff
@@ -0,0 +1,198 @@
+        setPropertyIfMetadataPresent(sequencedNode, JcrConstants.JCR_MIME_TYPE, mimeType);
+        return processBasicMetadata(sequencedNode, binaryValue);
+    }
+
+    private boolean processBasicMetadata( Node sequencedNode,
+                                          Binary binaryValue ) throws RepositoryException {
+        VideoMetadata metadata = null;
+        try (InputStream stream = binaryValue.getStream()) {
+            metadata = new VideoMetadata();
+            metadata.setInput(stream);
+            if (metadata.check()) {
+                setPropertyIfMetadataPresent(sequencedNode, DURATION, metadata.getDuration());
+                setPropertyIfMetadataPresent(sequencedNode, BITRATE, metadata.getBitrate());
+                setPropertyIfMetadataPresent(sequencedNode, TITLE, metadata.getTitle());
+                setPropertyIfMetadataPresent(sequencedNode, COMMENT, metadata.getComment());
+                setPropertyIfMetadataPresent(sequencedNode, ENCODER, metadata.getEncoder());
+
+                int suffix = 0;
+                for (StreamMetadata streamMetadata : metadata.getStreams()) {
+                    Node streamNode = sequencedNode.addNode(STREAM_NODE + String.valueOf(suffix), STREAM_NODE);
+                    processStreamMetadata(streamNode, streamMetadata);
+                    suffix += 1;
+                }
+                return true;
+            }
+        } catch (IOException e) {
+            getLogger().debug(e, "Couldn't process the stream.");
+        }
+        return false;
+    }
+
+    private boolean processStreamMetadata( Node streamNode,
+                                           StreamMetadata stream) throws RepositoryException {
+        setPropertyIfMetadataPresent(streamNode, STREAM_TYPE, stream.getStreamType());
+        setPropertyIfMetadataPresent(streamNode, CODEC, stream.getCodec());
+        setPropertyIfMetadataPresent(streamNode, BITRATE, stream.getBitrate());
+        setPropertyIfMetadataPresent(streamNode, FRAMERATE, stream.getFramerate());
+        setPropertyIfMetadataPresent(streamNode, SAMPLERATE, stream.getSamplerate());
+        setPropertyIfMetadataPresent(streamNode, CHANNELS, stream.getChannels());
+        setPropertyIfMetadataPresent(streamNode, WIDTH, stream.getWidth());
+        setPropertyIfMetadataPresent(streamNode, HEIGHT, stream.getHeight());
+
+        return true;
+    }
+
+    private Node getMetadataNode( Node outputNode ) throws RepositoryException {
+        if (outputNode.isNew()) {
+            outputNode.setPrimaryType(METADATA_NODE);
+            return outputNode;
+        }
+        return outputNode.addNode(METADATA_NODE, METADATA_NODE);
+    }
+
+    private void setPropertyIfMetadataPresent( Node node,
+                                               String propertyName,
+                                               Object value ) throws RepositoryException {
+        if (value != null) {
```

Comment:
see [this comment](https://github.com/ModeShape/modeshape/pull/1502#discussion_r48826407) for how to improve this chain of `instanceofs`